src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -147,10 +147,17 @@ protected function describeCallable(mixed $callable, array $options = []): void
         $this->writeData($this->getCallableData($callable), $options);
     }
 
-    protected function describeContainerParameter(mixed $parameter, array $options = []): void
+    protected function describeContainerParameter(mixed $parameter, ?array $deprecation, array $options = []): void
     {
         $key = $options['parameter'] ?? '';
 
+        if ($deprecation) {
+            $parameter = [
+                'value' => $parameter,
+                'deprecated' => sprintf('Since %s %s: %s', $deprecation[0], $deprecation[1], sprintf(...\array_slice($deprecation, 2))),
```

Comment:
```suggestion
                'deprecation' => sprintf('Since %s %s: %s', $deprecation[0], $deprecation[1], sprintf(...\array_slice($deprecation, 2))),
```

Reply:
Moved to an extra key `_deprecation`.